<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PDFEncryptionSettingsDialog</class>
 <widget class="QDialog" name="PDFEncryptionSettingsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>843</width>
    <height>620</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Encryption Settings</string>
  </property>
  <layout class="QVBoxLayout" name="dialogLayout">
   <item>
    <widget class="QGroupBox" name="encryptionMethodGroupBox">
     <property name="title">
      <string>Encryption Method</string>
     </property>
     <layout class="QGridLayout" name="methodGroupBoxLayout">
      <item row="0" column="2">
       <widget class="pdfviewer::PDFEncryptionStrengthHintWidget" name="algorithmHintWidget" native="true"/>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="encryptionAlgorithm">
        <property name="text">
         <string>Encryption algorithm</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="algorithmComboBox"/>
      </item>
      <item row="2" column="0" colspan="3">
       <widget class="QLabel" name="encryptionMethodHintLabel">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Select encryption algorithm. AES-256 is strongly recommended, because older encryption algorithm can be potentially broken. Select older algorithms (as AES-128 or RC4) only, if you need backward compatibility. Also, choose a strong password to ensure strong encryption.&lt;/p&gt;&lt;p&gt;Public key security using certificate is also supported. In that case, you must select a certificate with private key, and this certificate is then used to encrypt data. User, which wants to open document encrypted with certificate, must have a private key to the certificae.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="certificateLabel">
        <property name="text">
         <string>Certificate</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="certificateComboBox"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="passwordsGroupBox">
     <property name="title">
      <string>Passwords</string>
     </property>
     <layout class="QGridLayout" name="passwordsGroupBoxLayout">
      <item row="1" column="0">
       <widget class="QCheckBox" name="ownerPasswordEnableCheckBox">
        <property name="text">
         <string>Password (owner access)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QCheckBox" name="userPasswordEnableCheckBox">
        <property name="text">
         <string>Password (document open)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="userPasswordEdit">
        <property name="echoMode">
         <enum>QLineEdit::Password</enum>
        </property>
        <property name="clearButtonEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="pdfviewer::PDFEncryptionStrengthHintWidget" name="userPasswordStrengthHintWidget" native="true"/>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="ownerPasswordEdit">
        <property name="echoMode">
         <enum>QLineEdit::Password</enum>
        </property>
        <property name="clearButtonEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="pdfviewer::PDFEncryptionStrengthHintWidget" name="ownerPasswordStrengthHintWidget" native="true"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="encryptContentsGroupBox">
     <property name="title">
      <string>Encrypt Contents</string>
     </property>
     <layout class="QVBoxLayout" name="encryptGroupBoxLayout">
      <item>
       <widget class="QRadioButton" name="encryptAllRadioButton">
        <property name="text">
         <string>Encrypt all document contents, including document metadata</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="encryptAllExceptMetadataRadioButton">
        <property name="text">
         <string>Encrypt all document contets except metadata</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="encryptFileAttachmentsOnlyRadioButton">
        <property name="text">
         <string>Encrypt file attachments only</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="permissionsGroupBox">
     <property name="title">
      <string>Permissions</string>
     </property>
     <layout class="QGridLayout" name="permissionsGroupBoxLayout">
      <item row="0" column="0">
       <widget class="QCheckBox" name="permPrintLowResolutionCheckBox">
        <property name="text">
         <string>Print (low resolution)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="permFillInteractiveFormsCheckBox">
        <property name="text">
         <string>Fill interactive forms</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="permPrintHighResolutionCheckBox">
        <property name="text">
         <string>Print (high resolution)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="permAccessibilityCheckBox">
        <property name="text">
         <string>Accessibility</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="permModifyDocumentContentsCheckBox">
        <property name="text">
         <string>Modify document contents</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QCheckBox" name="permAssembleCheckBox">
        <property name="text">
         <string>Assemble document (insert, rotate, delete pages...)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="permInteractiveItemsCheckBox">
        <property name="text">
         <string>Edit interactive items (annotations, forms, ...)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QCheckBox" name="permCopyContentCheckBox">
        <property name="text">
         <string>Copy/extract document content</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>pdfviewer::PDFEncryptionStrengthHintWidget</class>
   <extends>QWidget</extends>
   <header>pdfencryptionstrengthhintwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>PDFEncryptionSettingsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PDFEncryptionSettingsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
